`ModuleNotFoundError ` when building an extension

## Description

In a project installing dependencies in `node_modules` without any hoisting (e.g. with [pnpm](https://pnpm.io)), `jupyter labextension build .` will throw `ModuleNotFoundError: Module not found: Error: Can't resolve 'style-loader' in '/Users/johndoe/project/javascript/jupyterlab-extension'`

<details>
<summary>full stack trace</summary>
<pre>
Building extension in .
│ Compilation starting…
│ ModuleNotFoundError: Module not found: Error: Can't resolve 'style-loader' in '/Users/johndoe/project/javascript/jupyterlab-extension'
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/Compilation.js:2022:28
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/NormalModuleFactory.js:817:13
│     at eval (eval at create (/Users/johndoe/project/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/NormalModuleFactory.js:275:22
│     at eval (eval at create (/Users/johndoe/project/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/NormalModuleFactory.js:560:15
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/NormalModuleFactory.js:118:11
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/NormalModuleFactory.js:631:8
│     at /Users/johndoe/project/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:2830:7
│     at done (/Users/johndoe/project/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:2925:13)
│ resolve 'style-loader' in '/Users/johndoe/project/javascript/jupyterlab-extension'
│   Parsed request is a module
│   using description file: /Users/johndoe/project/javascript/jupyterlab-extension/package.json (relative path: .)
│     resolve as module
│       looking for modules in /Users/johndoe/project/javascript/jupyterlab-extension/node_modules
│         single file module
│           using description file: /Users/johndoe/project/javascript/jupyterlab-extension/package.json (relative path: ./node_modules/style-loader)
│             no extension
│               /Users/johndoe/project/javascript/jupyterlab-extension/node_modules/style-loader doesn't exist
│             .js
│               /Users/johndoe/project/javascript/jupyterlab-extension/node_modules/style-loader.js doesn't exist
│         /Users/johndoe/project/javascript/jupyterlab-extension/node_modules/style-loader doesn't exist
│       /Users/johndoe/project/javascript/node_modules doesn't exist or is not a directory
│       looking for modules in /Users/johndoe/project/node_modules
│         single file module
│           using description file: /Users/johndoe/project/package.json (relative path: ./node_modules/style-loader)
│             no extension
│               /Users/johndoe/project/node_modules/style-loader doesn't exist
│             .js
│               /Users/johndoe/project/node_modules/style-loader.js doesn't exist
│         /Users/johndoe/project/node_modules/style-loader doesn't exist
│       /Users/tibdex/work/node_modules doesn't exist or is not a directory
│       /Users/tibdex/node_modules doesn't exist or is not a directory
│       /Users/node_modules doesn't exist or is not a directory
│       /node_modules doesn't exist or is not a directory
│ /Users/johndoe/project/node_modules/.pnpm/@jupyterlab+builder@4.0.7_@swc+core@1.3.94/node_modules/@jupyterlab/builder/lib/build-labextension.js:104
│             throw new Error(err.details);
│             ^
│ Error: resolve 'style-loader' in '/Users/johndoe/project/javascript/jupyterlab-extension'
│   Parsed request is a module
│   using description file: /Users/johndoe/project/javascript/jupyterlab-extension/package.json (relative path: .)
│     resolve as module
│       looking for modules in /Users/johndoe/project/javascript/jupyterlab-extension/node_modules
│         single file module
│           using description file: /Users/johndoe/project/javascript/jupyterlab-extension/package.json (relative path: ./node_modules/style-loader)
│             no extension
│               /Users/johndoe/project/javascript/jupyterlab-extension/node_modules/style-loader doesn't exist
│             .js
│               /Users/johndoe/project/javascript/jupyterlab-extension/node_modules/style-loader.js doesn't exist
│         /Users/johndoe/project/javascript/jupyterlab-extension/node_modules/style-loader doesn't exist
│       /Users/johndoe/project/javascript/node_modules doesn't exist or is not a directory
│       looking for modules in /Users/johndoe/project/node_modules
│         single file module
│           using description file: /Users/johndoe/project/package.json (relative path: ./node_modules/style-loader)
│             no extension
│               /Users/johndoe/project/node_modules/style-loader doesn't exist
│             .js
│               /Users/johndoe/project/node_modules/style-loader.js doesn't exist
│         /Users/johndoe/project/node_modules/style-loader doesn't exist
│       /Users/tibdex/work/node_modules doesn't exist or is not a directory
│       /Users/tibdex/node_modules doesn't exist or is not a directory
│       /Users/node_modules doesn't exist or is not a directory
│       /node_modules doesn't exist or is not a directory
│     at compilerCallback (/Users/johndoe/project/node_modules/.pnpm/@jupyterlab+builder@4.0.7_@swc+core@1.3.94/node_modules/@jupyterlab/builder/lib/build-labextension.js:104:19)
│     at /Users/johndoe/project/node_modules/.pnpm/@jupyterlab+builder@4.0.7_@swc+core@1.3.94/node_modules/@jupyterlab/builder/lib/build-labextension.js:146:21
│     at /Users/johndoe/project/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:2830:7
│     at done (/Users/johndoe/project/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:2865:11)
│     at /Users/johndoe/project/node_modules/.pnpm/neo-async@2.6.2/node_modules/neo-async/async.js:2818:7
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/HookWebpackError.js:68:3
│     at Hook.eval [as callAsync] (eval at create (/Users/johndoe/project/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
│     at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Users/johndoe/project/node_modules/.pnpm/tapable@2.2.1/node_modules/tapable/lib/Hook.js:18:14)
│     at Cache.shutdown (/Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/Cache.js:150:23)
│     at /Users/johndoe/project/node_modules/.pnpm/webpack@5.89.0_@swc+core@1.3.94_webpack-cli@5.1.4/node_modules/webpack/lib/Compiler.js:1229:15
</pre>
</details>

This is caused by [Webpack loaders not being declared with `require.resolve()`](https://stackoverflow.com/a/58854073).

## Reproduce

1. Create an new JupyterLab extension, using pnpm 8.x as the package manager
2. Run `pnpm run jupyter labextension build .`

## Expected behavior

The build succeeds.

## Context

- Operating System and version: macOS Sonoma
- Browser and version: N/A
- JupyterLab version: `main`

--- a/builder/src/build.ts
+++ b/builder/src/build.ts
@@ -205,7 +205,7 @@ export namespace Build {
           rules: [
             {
               test: /\.css$/,
-              use: [MiniCssExtractPlugin.loader, 'css-loader']
+              use: [MiniCssExtractPlugin.loader, require.resolve('css-loader')]
             },
             {
               test: /\.svg/,

--- a/builder/src/extensionConfig.ts
+++ b/builder/src/extensionConfig.ts
@@ -257,7 +257,7 @@ function generateConfig({
     rules.push({
       test: /\.js$/,
       enforce: 'pre',
-      use: ['source-map-loader']
+      use: [require.resolve('source-map-loader')]
     });
   }
 

--- a/builder/src/webpack.config.base.ts
+++ b/builder/src/webpack.config.base.ts
@@ -5,7 +5,10 @@ import miniSVGDataURI from 'mini-svg-data-uri';
 
 const rules = [
   { test: /\.raw\.css$/, type: 'asset/source' },
-  { test: /(?<!\.raw)\.css$/, use: ['style-loader', 'css-loader'] },
+  {
+    test: /(?<!\.raw)\.css$/,
+    use: [require.resolve('style-loader'), require.resolve('css-loader')]
+  },
   { test: /\.txt$/, type: 'asset/source' },
   { test: /\.md$/, type: 'asset/source' },
   { test: /\.(jpg|png|gif)$/, type: 'asset/resource' },

